Automattic/jetpack
# JITM: Fix button overflow when text length is too big

Follow-up to #29709
Fixes https://github.com/Automattic/jetpack/issues/29672#issuecomment-1484909960

## Proposed changes:

* Adds `overflow: hidden` and `flex-shrink` to make sure the button shrinks if there is not enough space, but shrinking the message horizontally first so that the text in the button remains intact until absolutely necessary.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






To view this specific JITM message:
* Purchase the Jetpack Complete plan, but do not apply it to any site
* Go to the My Jetpack page and wait a few seconds for the JITM to appear
* Resize the window and check that the `X` close button is not overflown
* Check that the button text is truncated with ellipsis when there is no more space left
* Apply a mobile view and select Galaxy Fold, if available, or 280x653px resolution
* Check that the button does not overflow anymore

![2023-03-28_11-42-10](https://user-images.githubusercontent.com/8486249/228278359-790ec1e2-7253-4a80-8ed8-bf0f7f32ab9b.png)
![2023-03-28_11-41-25](https://user-images.githubusercontent.com/8486249/228278367-f45b80f0-28c8-4bc1-8693-55648f8031b1.png)

https://user-images.githubusercontent.com/8486249/228278516-0091bd06-c678-4314-9f58-47835251140f.mp4
